concourse/scripts/extract_tf_vars_from_yaml.rb

```diff
@@ -23,7 +23,7 @@ def process_yaml(yaml_tree, prefix_chain)
       process_yaml(v, prefix_chain + [name])
     }
   else
-    puts "#{prefix_chain.join('_')}='#{yaml_tree}'"
+    puts "#{prefix_chain.join('_').tr('-', '_')}='#{yaml_tree}'"
```

Comment:
I wonder whether we should substitute all characters that aren't valid in shell variable names here...

Also, I think it's getting to the point where it's worth adding tests for this script.

Reply:
> Also, I think it's getting to the point where it's worth adding tests for this script.

You know that I would also say that is the point to be more smart handling secrets and variables, rather than parsing and converting `.yml` files :) 

I did the very minimum. TBH the ruby script change is not needed anymore because we do not consume this file in terraform destroy from the other commit. but it was a simple change.

I think if we enable credhub, which seems something we will do soon, we will refactor how we do all this stuff. Being more ambitious, if we start deploying other tech stacks for containers, we might have to solve how to share secrets&variables, and likely we will refactor all of this too... 

So I wouldn't really add a test for this

Reply:
This unbroke stuff for me and looks legit. I'm going to merge this because it will stop terraform destroy from breaking.

Hector's suggestions seem sensible to the novice eye.